CONTRIBUTING.md

```diff
@@ -7,94 +7,120 @@ Start small. The PRs most likely to be merged are the ones that make small,
 easily reviewed changes with clear, and specific intentions. See below for more
 [guidelines on pull requests](#pull-requests).
 
-Stick to issues that are
-[on the roadmap](https://github.com/WhisperSystems/Signal-Desktop/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22On+the+roadmap%22).
-Issues that are not included in this milestone may be not yet triaged,
-unplanned, or not actionable for one reason or another.
+It's a good idea to gauge interest in your intended work by finding the current issue
+for it or creating a new one yourself. You can use also that issue as a place to signal
+your intentions, and get feedback from the users most likely to appreciate your changes.
 
-If you start working on an issue, leave a comment to let others know. It is
-also a good idea to outline your approach to the problem in order to get
-feedback.
+Once you've spent a little bit of time planning your solution, it's a good idea to go
+back to the issue and talk about your approach. We'd be happy to provide feedback. [An
+ounce of prevention, as they say!](https://www.goodreads.com/quotes/247269-an-ounce-of-prevention-is-worth-a-pound-of-cure)
 
 ## Developer Setup
-```
-git clone https://github.com/WhisperSystems/Signal-Desktop.git
-cd Signal-Desktop
-npm install            # install dependencies
-npm start              # run
-```
 
-Some features, such as native notifications, will not work locally until you
-package for your platform:
+First, you'll need Node.js installed. Anything 6.x or newer works as of Summer 2017.
+You might consider a [version](https://github.com/creationix/nvm) [manager](https://github.com/coreybutler/nvm-windows) to make this easier for yourself.
+
+Then you need `git`, if you don't have that yet: https://git-scm.com/
+
+And for the final step before we actually get started, you'll need build tools to install
+the native modules used by the app. On Windows, it's easiest to open the Command Prompt
+(`cmd.exe`) as Administrator (right-click) and run this:
 
 ```
-npm run pack-staging   # build a packaged app dir in ./dist
+npm install --global --production windows-build-tools
 ```
+
+On OSX you can install the [XCode Command-line tools](http://osxdaily.com/2014/02/12/install-command-line-tools-mac-os-x/). On Linux you'll need to take a trip to your
+favorite package manager. Python 2.x and GCC two key necessary components.
+
+Now, run these commands in your preferred terminal:
+
 ```
-npm run dist-staging   # build installer (.dmg, .exe, or .deb)
+git clone https://github.com/WhisperSystems/Signal-Desktop.git
+cd Signal-Desktop
+npm install -g yarn    # (only if you don't already have yarn)
+yarn install           # install and build dependencies
+yarn run pack prod     # build dependencies correctly for electron environment
```

Comment:
pack prod should be pack-prod, but actually I think if we add `"postinstall": "electron-builder install-app-deps"` to our package.json scripts, we can drop this step because `yarn install` will then call electron-builder to rebuild dependencies, and it's faster than running the whole packaging process. :tada:

Reply:
Awesome. When looking at the command-line options, I was distracted by the `node-gyp-rebuild` option, which didn't work for me.